### ISSUE TITLE
RuntimeWarning: subpackages can technically be lazily loaded

### ISSUE THREAD

**Issue by jppyykm** [pyjanitor-devs/pyjanitor#1276]
I am running a script and my solution worked just fine but I got this warning. I wanted to inform you. If you need any more information, please let me know.

Script is  eagerly loaded even if it is already lazily loaded

- Operating system: Windows 10 
- Python version 3.11

```
import janitor
import pandas as pd
from datetime import datetime
import pandas as pd 

beginning = pd.to_datetime('14/09/2021', dayfirst=True)
today = pd.to_datetime(datetime.today())
date_range = pd.date_range(start=beginning, end=today)
date_range = pd.Series(date_range, name = 'date')

block = pd.DataFrame(
    {'Identifier': ['4913151F', 'F4E9124A', '31715888', 'D0C57FCA', '57B4D7EB', 'E46F1E5D', '99E0A2F8', 'D77E342E',
                    'C596D233', 'D0EED63F', 'D0C57FCA'],
     'start_date': ['03/11/2020', '05/07/2022', '22/12/2016', '17/03/2024', '14/10/2022', '08/08/2022', '04/11/2020',
                    '13/03/2023', '05/11/2021', '12/27/2022', '13/06/2022'],
     'end_date': ['11/07/2023', '11/04/2023', '14/12/2018', '20/01/2025', '15/06/2023', '09/01/2023', '16/07/2022',
                  '19/05/2024', '24/09/2022', '17/11/2023', '13/06/2023'],
     'color': ['red', 'green', 'magenta', 'yellow', 'light_blue', 'dark_blue', 'black', 'white', 'pink', 'orange',
               'yellow']})

block = block.conditional_join(date_range, ('start_date', 'date' '<='),
            ('end_date', 'date', '>='), use_numba=False)
print(block)
```

# Error Messages
```
<!--  \venv\Lib\site-packages\lazy_loader_init_.py:185: RuntimeWarning: subpackages can technically be lazily loaded, but it causes the package to be eagerly loaded even if it is already lazily loaded.So, you probably shouldn't use subpackages with this lazy feature. warnings.warn(msg, RuntimeWarning) -->
```

**Comment by jppyykm** [comment]
How can I ignore this warning?

**Comment by samukweku** [comment]
@ericmjl @thatlittleboy any ideas?

**Comment by samukweku** [comment]
thanks @jppyykm for the feedback. If you use a lower python version (3.8/3.9) do you get an error message? if you can, kindly create a new environment with a lower python version, and see if the error pops up again

**Comment by ericmjl** [comment]
Hmmm, I think we should just get rid of lazy loading. It was originally implemented out of a desire for speed, but I'm seeing a tradeoff. 

I will work on this matter over the coming few days.

**Comment by jppyykm** [comment]
Of course. I ran it in Python 3.9 and got the same warning. I tried silencing it but couldn't, does anyone know how to do so?

**Comment by samukweku** [comment]
@jppyykm there is a pending PR that should fix this. in the mean time, you can use import warnings to silence it

**Comment by jppyykm** [comment]
@samukweku Yes! Very grateful. I tried that and it doesn't work. Not sure it's relevant but it is perhaps more information to consider. Thanks!

**Comment by samukweku** [comment]
@jppyykm So the code below doesn't work?

```
import warnings
warnings.simplefilter("ignore", RuntimeWarning)
```

**Comment by jppyykm** [comment]
@samukweku It does not. It has no effect. 

I found a way to do so:

```
with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    import janitor
```

**Comment by samukweku** [comment]
#1278 fixes this. @jppyykm kindly reinstall the Dev version.

**Comment by ericmjl** [comment]
@jppyykm I have cut a new minor release. Give the bots a few minutes and we should be able to see the releases out in the wild.

**Comment by samukweku** [comment]
@ericmjl it seems the release failed? https://github.com/pyjanitor-devs/pyjanitor/actions/runs/5649973510/job/15305429936#step:13:72

**Comment by ericmjl** [comment]
Yeah it did. I’m gonna look at it after hours, doing kids stuff right now. Thanks for the heads up though.

**Comment by samukweku** [comment]
@ericmjl I feel it should be a major release cos of some possibly breaking changes

**Comment by ericmjl** [comment]
@samukweku I agree with you. That said, we are pre-1.0, so minor == major for us, and patch == minor.

**Comment by samukweku** [comment]
@jppyykm closing this - it has been fixed with #1278; you should not have this issue with pyjanitor version 0.25. feel free to give more feedback if you still have this issue. thanks
